src/client.rs

```diff
@@ -266,6 +269,47 @@ impl Client {
         return Ok(result);
     }
 
+    /// Get a key from memcached server and deserialize it as JSON. This function is only available when the `json` feature is enabled.
+    /// The value type should implement `FromMemcacheValueExt` and `serde::Deserialize`.
+    ///
+    /// Example:
+    /// ```rust
+    /// use serde::{Serialize, Deserialize};
+    ///
+    /// #[derive(Debug, Serialize, Deserialize)]
+    /// struct MyStruct {
+    ///     value: String,
+    ///     number: i32,
+    ///     flag: bool,
+    ///     array: Vec<i32>,
+    /// }
+    ///
+    /// let client = memcache::Client::connect("memcache://localhost:12345").unwrap();
+    /// let value = MyStruct {
+    ///     value: "hello".to_string(),
+    ///     number: 42,
+    ///     flag: true,
+    ///     array: vec![1, 2, 3],
+    /// };
+    /// client.set_json("foo", value, 10).unwrap();
```

Comment:
This tests the `set_json` function below as well, ensuring that the value is actually set correctly